### DISCUSSION TITLE
使用torch.compile()一行优化

### DISCUSSION THREAD

**Discussion by hyskylord** [Equim-chan/Mortal#43] [General]
最近尝试了pytorch 2.X的compile功能，使用后训练速度大概提升了10%。一个小的遗憾是test_play由于目前输入输出的batch size不固定暂时无法优化。

Pytorch版本：2.1.0.dev20230315 py3.10_cuda11.8_cudnn8.7.0_0

mode：
mortal = torch.compile(mortal, mode="reduce-overhead")
current_dqn = torch.compile(current_dqn, mode="reduce-overhead")
next_rank_pred = torch.compile(next_rank_pred, mode="reduce-overhead")

效果：~6.8batch/s → ~7.5 batch/s

**Comment by Equim-chan** [comment]
之前在我的环境里 pytorch 2 在 inference 的时候有机率 coredumps（和是否使用 torch.compile 无关），就先回滚回了 1。

**Comment by Equim-chan** [comment]
v4 会包含一个配置项来开启 compile。
